app/src/main/java/com/weatherxm/usecases/ForecastUseCaseImpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.weatherxm.usecases
+
+import arrow.core.Either
+import com.weatherxm.data.ApiError
+import com.weatherxm.data.Failure
+import com.weatherxm.data.HourlyWeather
+import com.weatherxm.data.network.ErrorResponse.Companion.INVALID_TIMEZONE
+import com.weatherxm.data.repository.WeatherForecastRepository
+import com.weatherxm.ui.common.UIDevice
+import com.weatherxm.ui.common.UIForecast
+import com.weatherxm.ui.common.UIForecastDay
+import java.time.ZoneId
+import java.time.ZonedDateTime
+
+@Suppress("LongParameterList")
+class ForecastUseCaseImpl(
+    private val weatherForecastRepository: WeatherForecastRepository
+) : ForecastUseCase {
+
+    @Suppress("MagicNumber")
+    override suspend fun getForecast(
+        device: UIDevice,
+        forceRefresh: Boolean
+    ): Either<Failure, UIForecast> {
+        if (device.timezone.isNullOrEmpty()) {
+            return Either.Left(ApiError.UserError.InvalidTimezone(INVALID_TIMEZONE))
+        }
+        val nowDeviceTz = ZonedDateTime.now(ZoneId.of(device.timezone))
```

Comment:
IMO is better to get the timezone from the forecast response instead of injecting the device object

Reply:
This is true but we need this `nowDeviceTz` in order to create the `dateEnd` for the API call below:
```
        val nowDeviceTz = ZonedDateTime.now(ZoneId.of(device.timezone))
        val dateEndInDeviceTz = nowDeviceTz.plusDays(7)
        return weatherForecastRepository.getDeviceForecast(
            device.id,
            nowDeviceTz,
            dateEndInDeviceTz,
            forceRefresh
        )
```
So we need the timezone before we have the actual response that's why we inject the device object.